harness/determined/common/api/_session.py

```diff
@@ -100,13 +158,102 @@ def put(
     ) -> requests.Response:
         return self._do_request("PUT", path, params, json, data, headers, timeout, False)
 
+
+class UnauthSession(BaseSession):
```

Comment:
I don't think of myself as an overengineering type. I really like this class hierarchy.

Reply:
thank you